packages/pg/lib/connection-parameters.js

```diff
@@ -33,6 +34,27 @@ var readSSLConfigFromEnvironment = function () {
   return defaults.ssl
 }
 
+var fillInSSLVariablesFromEnvironment = function(existingSsl) {
+  var caFileName = val('sslrootcert', existingSsl)
+  var crtFileName = val('sslcert', existingSsl)
+  var keyFileName = val('sslkey', existingSsl)
+
+  // Only try to load fs if we expect to read from the disk
+  const fs = caFileName || crtFileName || keyFileName ? require('fs') : null
+  var result = {}
+  if (caFileName) {
+    result.ca = fs.readFileSync(caFileName).toString()
```

Comment:
Implicit sync calls that happen on every connection might not be good (one good reason for user code to be responsible for this…)

Reply:
Regarding sync / async, read this: https://github.com/brianc/node-postgres/pull/2994#discussion_r1218162821